src/cogent3/core/new_alignment.py

```diff
@@ -357,6 +369,101 @@ def _(seq: numpy.ndarray, moltype: MolType) -> tuple[str, IndelMap]:
     return ungapped, indel_map
 
 
+@singledispatch
+def prep_for_seq_data(
+    data: Union[dict, list, numpy.ndarray], alphabet: CharAlphabet
+) -> dict[str, numpy.ndarray]:
+    """Assuming data is a series of seqs, a series of[name, seq] pairs, or a
+    dict {name: seq}"""
+    raise NotImplementedError(
+        f"{prep_for_seq_data.__name__} not implemented for type {type(data)}"
+    )
+
+
+@prep_for_seq_data.register
+def _(data: dict, alphabet: CharAlphabet) -> dict[str, numpy.ndarray]:
+    # todo: kath, apply label to name here
+    return {str(k): seq_index(v, alphabet=alphabet) for k, v in data.items()}
+
+
+@prep_for_seq_data.register
+def _(data: list, alphabet: CharAlphabet) -> dict[str, numpy.ndarray]:
+    # if data = [seq, ...]
+    if isinstance(data[0], (str, bytes, numpy.ndarray)):
+        names = assign_sequential_names(len(data))
+        named_seqs = dict(zip(names, data))
+        return prep_for_seq_data(named_seqs, alphabet=alphabet)
+
+    # if data = [[name, seq], ...] or [(name, seq), ...]
+    elif isinstance(data[0], (list, tuple)):
+        named_seqs = dict(data)
+        return prep_for_seq_data(named_seqs, alphabet=alphabet)
+
+
+@prep_for_seq_data.register
+def _(data: numpy.ndarray, alphabet: CharAlphabet) -> dict[str, numpy.ndarray]:
+    names = assign_sequential_names(data.shape[0])
+    named_seqs = dict(zip(names, data))
+    return prep_for_seq_data(named_seqs, alphabet=alphabet)
+
+
+def make_unaligned_seqs(
+    data: Union[dict, list, numpy.ndarray],
+    moltype: Union[str, MolType],
+    label_to_name: Callable = None,
+    info: dict = None,
+    source: Union[str, Path] = None,
+    **kwargs,
+):
+    """Initialize an unaligned collection of sequences.
+
+    Parameters
+    ----------
+    data
```

Comment:
let's decide what forms of `data` we are going to support being passed to `make_unaligned_seqs`. currently I am supporting: 
- `dict{name: seq, ...}`
- `[[name, seq]...]` (lists or sets) 
- `[seq, ...]` (lists or sets) 
- `2D numpy array`
which of these are actually necessary and which have I missed?

Reply:
Should we support creating a `SequenceCollection` where data is already a `SeqData` instance?